src/util/transition/ngbTransition.ts

```diff
@@ -0,0 +1,48 @@
+import {EMPTY, fromEvent, Observable, of, Subject} from 'rxjs';
+import {take} from 'rxjs/operators';
+
+export interface NgbTransitionOptions {
+  animation: boolean;
+  running: 'continue' | 'restart';
+}
+
+const runningTransitions = new Map<HTMLElement, Subject<any>>();
+
+export const ngbRunTransition =
+    (element: HTMLElement, startFn: (element: HTMLElement) => void, options: NgbTransitionOptions):
+        Observable<undefined> => {
+
+          // checking if animations are enabled
+          if (!options.animation) {
+            return of(undefined);
```

Comment:
FYI, the difference between `of()` and `EMTPY` is that both are completed, but of also returns a value before completion. See unit tests for more details.